src/App.jsx

```diff
@@ -0,0 +1,30 @@
+import React, {useState} from "react";
+import { BrowserRouter as Router, Redirect, Switch, Route } from "react-router-dom"
+import { SpotifyLogin } from "./SpotifyLogin";
+import { SpotifyAuthRedirect } from "./SpotifyAuthRedirect";
+import { SpotifyPlaylists } from "./SpotifyPlaylists";
+
+export const App = () => {
+    const [loggedIn, setLoggedIn] = useState(false);
+
+    return (
+        <Router>
+            <Switch>
+                <Route exact path="/">
+                    {
+                        loggedIn ?
```

Comment:
There is no need to mutually exclude those 2 components. I'm suggesting to:
- show Login button at page header if not logged it, if user logged in -> show logged in username instead of a button
- always show SpotifyPlaylist, but disable interaction if user not logged it